lib/i18n/backend/base.rb

```diff
@@ -236,14 +236,23 @@ def load_rb(filename)
           eval(IO.read(filename), binding, filename)
         end
 
+        # Feature tests YAML load options and only selects those that
+        # are supported in the current version of Psych.
+        YAML_LOAD_OPTIONS = { symbolize_names: true, freeze: true }.select do |k, v|
```

Comment:
Rather than rescuing on an `ArgumentError`, I would rather a version check (either of Ruby, or of whatever's added the support for these options). 

Pseudocode like:

```
if version >= 3.0.0
  <symbolise code>
else
  <old path>
end
```

Reply:
The problem is that bundled gem sometimes "lie" about their version because of https://bugs.ruby-lang.org/issues/18169, it happened to me previously hence I I now default to feature testing.

That being said, we could conflate this with the feature testing for `unsafe_load_file`:

```ruby
            if YAML.respond_to?(:unsafe_load_file) # Psych 4.0 way
              YAML.unsafe_load_file(filename, symbolize_names: true, freeze: true)
            else
              YAML.load_file(filename)
            end
```

We'll miss a few versions but that's probably fine.